markets/perps-market/contracts/storage/KeeperCosts.sol

```diff
@@ -0,0 +1,135 @@
+            price.slot := s
+        }
+    }
+
+    function update(Data storage self, bytes32 feedId) internal {
+        self.feedId = feedId;
+    }
+
+    function getSettlementKeeperCosts(
+        Data storage self,
+        uint128 accountId
+    ) internal view returns (uint sUSDCost) {
+        PerpsMarketFactory.Data storage factory = PerpsMarketFactory.load();
+
+        accountId; // unused for now, but will be used to calculate rewards based on account collaterals in the future
+
+        (bytes32[] memory runtimeKeys, bytes32[] memory runtimeValues) = getRuntime(
+            0,
+            0,
+            KIND_SETTLEMENT
+        );
+
+        sUSDCost = INodeModule(factory.oracle)
+            .processWithRuntime(self.feedId, runtimeKeys, runtimeValues)
+            .price
+            .toUint();
+    }
+
+    function getFlagKeeperCosts(
+        Data storage self,
+        uint128 accountId
+    ) internal view returns (uint sUSDCost) {
+        PerpsMarketFactory.Data storage factory = PerpsMarketFactory.load();
+
+        PerpsAccount.Data storage account = PerpsAccount.load(accountId);
+        uint numberOfUpdatedFeeds = account.getNumberOfCollaterals(false) +
+            account.openPositionMarketIds.length();
+
+        (bytes32[] memory runtimeKeys, bytes32[] memory runtimeValues) = getRuntime(
+            0,
+            numberOfUpdatedFeeds,
+            KIND_FLAG
+        );
+
+        sUSDCost = INodeModule(factory.oracle)
+            .processWithRuntime(self.feedId, runtimeKeys, runtimeValues)
+            .price
+            .toUint();
+    }
+
+    function getLiquidateKeeperCosts(Data storage self) internal view returns (uint sUSDCost) {
+        PerpsMarketFactory.Data storage factory = PerpsMarketFactory.load();
+
+        (bytes32[] memory runtimeKeys, bytes32[] memory runtimeValues) = getRuntime(
+            0,
+            0,
+            KIND_LIQUIDATE
+        );
+
+        sUSDCost = INodeModule(factory.oracle)
+            .processWithRuntime(self.feedId, runtimeKeys, runtimeValues)
+            .price
+            .toUint();
+    }
+
+    function getTotalFlagAndLiquidationCost(
+        Data storage self,
+        uint128 accountId,
+        uint256 numberOfChunks
+    ) internal view returns (uint sUSDCost) {
+        PerpsMarketFactory.Data storage factory = PerpsMarketFactory.load();
+
+        PerpsAccount.Data storage account = PerpsAccount.load(accountId);
+        uint numberOfUpdatedFeeds = account.activeCollateralTypes.length() +
+            account.openPositionMarketIds.length();
+
+        (bytes32[] memory runtimeKeys, bytes32[] memory runtimeValues) = getRuntime(
+            numberOfChunks,
+            numberOfUpdatedFeeds,
+            KIND_LIQUIDATION_ELIGIBILITY
+        );
+
+        sUSDCost = INodeModule(factory.oracle)
+            .processWithRuntime(self.feedId, runtimeKeys, runtimeValues)
+            .price
+            .toUint();
+    }
+
+    function getRuntime(
```

Comment:
```suggestion
    function _processWithRuntime(
```

seems like everytime `getRuntime` is called, we usually call process so can just move it all into this private method